cupyx/scipy/sparse/csr.py

```diff
@@ -295,8 +295,10 @@ def transpose(self, axes=None, copy=False):
                 'swapping dimensions is the only logical permutation.')
 
         shape = self.shape[1], self.shape[0]
-        return csc.csc_matrix(
+        trans = csc.csc_matrix(
             (self.data, self.indices, self.indptr), shape=shape, copy=copy)
+        trans._has_canonical_format = self._has_canonical_format
+        return trans
```

Comment:
I am still concerned in this transpose operation. If [I understand it correctly](https://github.com/cupy/cupy/pull/3430#discussion_r439567412), `self.indices` and `self.indptr` should be updated during transpose. Am I missing something?

Reply:
I think for the transpose operation as used in e.g. csrmv in cupy/cusparse.py ...cusparse.spMV(), you  compute  the multiplication directly with the transposed matrix using a _transpose_flag - without actually computing the transpose.

Reply:
It seems SciPy does this too. I don't understand...

Reply:
We left comments simultaneously, @smarkesini! It's good to know `_transpose_flag` -- I wasn't aware of that. But the `transpose` method here is for high-level operations, say, `b=a.T`.

Reply:
Generally the sparse matrix operations follow the [sparse BLAS](https://math.nist.gov/spblas/) conventions, with transpose operations computed internally, so to get b=a.T you can just update the flag without sorting everything again; then as long as any operation applied to b is aware of the flag everything works.

Reply:
I dug deeper into SciPy, and it turns out that for CSR:
1. `tocsc` will ensure the indices to be sorted, but the outcome is not necessary in the canonical form (i.e., can have repeated indices)
2. `transpose` will ensure nothing.

This observation also holds vice versa for CSC. So, the change here in `csr.py` is not necessary, and `csc.py` will need to be fixed.

Reply:
> so to get b=a.T you can just update the flag without sorting everything again; then as long as any operation applied to b is aware of the flag everything works

Right, that's what I expected, but I dunno where this flag is...😅

Reply:
I guess `shape` might be used for this purpose...?